### PR TITLE
feat: Added Layout Screen for Quiz Result.

### LINKED ISSUE #153: Create Quiz Result Screen
References
Screenshot 2019-05-14 at 1 33 41 PM

### PR DESCRIPTION
**Pull request by thedevelopersanjeev**
Fixes #153 

Changes: Added Layout for Quiz Result Screen using QuizResultFragment. Also, fixed the intent to launch this screen after the Quiz has ended.

Screenshots for the change:
cbquizresult

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
You have to show appropriate result as well not just hard-coded numbers

**Comment by thedevelopersanjeev** [comment]
How can I get the total number of questions and correct answers from attemptId or is there any other way? 
Please provide a link to the documentation of API if there is any.
@aggarwalpulkit596

**Comment by aggarwalpulkit596** [comment]
@thedevelopersanjeev there isn't documentation as of now you can either check on the web otherwise you can get it by v2/quiz_attempts/{quizId}

**Comment by thedevelopersanjeev** [comment]
@aggarwalpulkit596  
private const val ARG_QUIZ_ID = "quiz_id"
private const val ARG_ATTEMPT_ID = "attempt_id"
private const val ARG_QUIZ_ATTEMPT_ID = "quiz_attempt_id"
which ID is going to return me the QuizAttempt? I have tried both attempt_id and quiz_attempt_id.
both of them are returning null.
`Clients.onlineV2JsonApi.getQuizAttemptById(attemptId).enqueue(retrofitCallback { _, response ->
            val attempt = response?.body()
            // Log.v("Attempt : ", attempt.toString())
            val result = attempt?.result
            val score = result?.score
            val total = result?.questions?.size
            Log.v("Total : ", total.toString())
            Log.v("Score : ", score.toString())
        })`
This is the code that I am using to retrieve total number of question and score of the quiz.

**Comment by aggarwalpulkit596** [comment]
Check cb website you'll get it from there which end point you need and what query params are needed

**Comment by thedevelopersanjeev** [comment]
Please check my last commit @aggarwalpulkit596 
Quiz Result is coming as null in the response body and the creation time is also wrong
I have added the logcat in comments... 
Line 38 in QuizResultFragment.kt file...

**Comment by aggarwalpulkit596** [comment]
quiz result will give null if the quiz hasn't been submitted yet.
Did you check the website is it working there?

**Comment by thedevelopersanjeev** [comment]
I have tried submitting the quiz using and then query score and total number of questions, but it is returning score as 0 and number of questions 1 in response...
also, I am looking for exact call which is made in web app here in this repository https://github.com/coding-blocks/codingblocks.online.projectx
but, I can't find the exact file.
Can you help  @aggarwalpulkit596 ?

**Comment by aggarwalpulkit596** [comment]
Will look into this tomorrow till then you can work on some other issue or may open some issues if you find them

**Comment by aggarwalpulkit596** [comment]
@thedevelopersanjeev the issue with the null result has been resolved it should get you the result now

**Comment by thedevelopersanjeev** [comment]
Yes, it is working fine now.
and I have updated the PR. Please review @aggarwalpulkit596 
Also, I think the quiz submit button should not be there as it is not visible and a user may end the quiz directly without submitting it, which was exactly happening with me.

I have created another issue with more description here 
https://github.com/coding-blocks/CBOnlineApp/issues/175

**Comment by aggarwalpulkit596** [comment]
Add Screenshot

**Comment by thedevelopersanjeev** [comment]
@aggarwalpulkit596 
Screenshot_20190517-031936

**Comment by aggarwalpulkit596** [comment]
Use black and green circle for total and correct questions respectively

**Comment by aggarwalpulkit596** [comment]
Card text are not properly Aligned and is button is also not centre aligned

**Comment by thedevelopersanjeev** [comment]
Updated Screenshot
Screenshot_20190517-045138

File: app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt [hunk 1]
```diff
@@ -155,13 +156,14 @@ class QuizFragment : Fragment(), AnkoLogger, ViewPager.OnPageChangeListener, Vie
...
         when (v.id) {
             R.id.nextBtn -> if (nextBtn.text == "End") {
-                val fragmentManager = fragmentManager!!
-                val fragmentTransaction = fragmentManager.beginTransaction()
-                fragmentTransaction.setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
-                fragmentTransaction.replace(R.id.framelayout_quiz,
-                        AboutQuizFragment.newInstance(quizId,
-                                attemptId,qnaId), "quiz")
-                fragmentTransaction.commit()
+                Clients.onlineV2JsonApi.sumbitQuizById(quizAttemptId).enqueue(retrofitCallback { throwable, response ->
```
[the remark this answers is not in the document]
Comment (thedevelopersanjeev) [reply app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt]: I have created another issue for this, here I would suggest to create a dialog and show a warning that the quiz is going to end.
I will change this in that issue.

[the remark this answers is not in the document]
Comment (thedevelopersanjeev) [reply app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt]: Also, I need to know, how I am going to pass quizAttemptId in submitButton onClick in ViewPager and replace the fragment there?

**Comment by thedevelopersanjeev** [comment]
fixed.
@aggarwalpulkit596

**Comment by thedevelopersanjeev** [comment]
Build is failing due to below error
```
com.codingblocks.onlineapi.api.OnlineJsonApiAuthenticatedTest > GET Quiz FAILED
    java.net.SocketTimeoutException at OnlineJsonApiAuthenticatedTest.kt:82
        Caused by: java.net.SocketException at OnlineJsonApiAuthenticatedTest.kt:82
14 tests completed, 1 failed
> Task :onlineapi:test FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':onlineapi:test'.
```

**Comment by aggarwalpulkit596** [comment]
Please use color shade which are more elegant to look 😅

**Comment by thedevelopersanjeev** [comment]
Using the color codes same as website @aggarwalpulkit596 
Screenshot_20190517-214256

**Pull request merged**